src/formats/onenote.ts

```diff
@@ -357,17 +385,108 @@ export class OneNoteImporter extends FormatImporter {
 		return pageElement;
 	}
 
+	/**
+	 * Returns a filesystem path for any OneNote entity like sections or notes
+	 * Paths are returned in the following format:
+	 * (Export folder)/Notebook/(possible section groups)/Section/(possible pages with a higher level) 
+	 */
+	getEntityPath(entityID: string, currentPath: string, parentEntity?: OnenoteEntityHierarchyModel | undefined): string | null {
+		let returnPath: string | null = null;
+		if (!parentEntity) {
+			for (const notebook of this.notebooks) {
+				const path = this.getEntityPath(entityID, `${currentPath}/${notebook.displayName}`, notebook);
+				if (path) {
+					returnPath = path;
+					break;
+				}
+			}
+		}
+		else {
+			if (parentEntity.id === this.currentSection) {
```

Comment:
This doesn't seem like the right place for this to be happening. Retrieving the entity path should not have a side-effect of updating the object state.